tools/adventure-pack/src/__tests__/iteratorPrototypeFilter.test.ts

```diff
@@ -0,0 +1,69 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "../iteratorPrototypeFilter";
+
+describe("Iterator.prototype.filter", () => {
+  it.each([
+    {
+      array: [2, 4, 6, 7, 8, 9],
+      condition: (p: number) => p % 2 === 0,
+      expected: [2, 4, 6, 8],
+    },
+    {
+      array: [1, 2, 3, 4, 1, 5],
+      condition: (p: number) => p > 2,
+      expected: [3, 4, 5],
+    },
+  ])("can filter an Array's values()", ({ array, condition, expected }) => {
+    const filterResult = array.values().filter(condition);
+
+    expected.forEach((e) => {
+      expect(filterResult.next().value).toBe(e);
+    });
```

Comment:
Let's use a regular for loop if we don't need the index that `forEach` provides?